File: backend/src/main/java/com/donatecommerce/service/CampaignService.java
package com.donatecommerce.service;

import java.util.List;
import java.util.UUID;

import com.donatecommerce.dto.request.CampaignCreateRequest;
import com.donatecommerce.dto.response.CampaignResponse;

public interface CampaignService {
    
    List<CampaignResponse> getAllCampaigns();
    
    List<CampaignResponse> getActiveCampaigns();
    
    List<CampaignResponse> getOngoingCampaigns();
    
    CampaignResponse getCampaignById(UUID id);
    
    CampaignResponse createCampaign(CampaignCreateRequest request);
    
    CampaignResponse updateCampaign(UUID id, CampaignCreateRequest request);
    
    void deleteCampaign(UUID id);
}

File: frontend/src/services/product/campaignService.ts
import { apiClient } from '../api/apiClient';
import { 
  CampaignResponse, 
  CampaignCreateRequest, 
  ApiResponse 
} from '../api/types';

export const campaignService = {
  getAllCampaigns: async (): Promise<CampaignResponse[]> => {
    return apiClient.get<CampaignResponse[]>('/api/campaigns');
  },

  getActiveCampaigns: async (): Promise<CampaignResponse[]> => {
    return apiClient.get<CampaignResponse[]>('/api/campaigns/active');
  },

  getCampaignById: async (id: string): Promise<CampaignResponse> => {
    return apiClient.get<CampaignResponse>(`/api/campaigns/${id}`);
  },

  createCampaign: async (campaign: CampaignCreateRequest): Promise<CampaignResponse> => {
    return apiClient.post<CampaignResponse>('/api/campaigns', campaign);
  },

  updateCampaign: async (id: string, campaign: CampaignCreateRequest): Promise<CampaignResponse> => {
    return apiClient.put<CampaignResponse>(`/api/campaigns/${id}`, campaign);
  },

  deleteCampaign: async (id: string): Promise<void> => {
    await apiClient.delete<void>(`/api/campaigns/${id}`);
  }
};
